File: app/src/main/java/com/futuretech/nfmovies/Adapter/MovieRecyclerViewAdapter.java
package com.futuretech.nfmovies.Adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.futuretech.nfmovies.Entity.Movie;
import com.futuretech.nfmovies.Listener.MovieItemClickListener;
import com.futuretech.nfmovies.R;
import com.futuretech.nfmovies.Utils.ImageUtil;

import java.util.List;

public class MovieRecyclerViewAdapter extends RecyclerView.Adapter<MovieRecyclerViewAdapter.ViewHolder> {
    private Context context;
    private List<Movie> movies;
    MovieItemClickListener movieItemClickListener;
    public MovieRecyclerViewAdapter(Context context, List<Movie> movies, MovieItemClickListener listener) {
        this.context = context;
        this.movies = movies;
        this.movieItemClickListener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.movie_item, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.name.setText(movies.get(position).getName());
        ImageUtil.display(context, movies.get(position).getImg(), holder.img, null);
    }

    @Override
    public int getItemCount() {
        return movies.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private TextView name;
        private ImageView img;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.movie_name);
            img = itemView.findViewById(R.id.movie_img);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    movieItemClickListener.onMovieClick(movies.get(getAdapterPosition()), img);
                }
            });
        }
    }
}

File: app/src/main/java/com/futuretech/nfmovies/Adapter/MovieRecyclerViewAdapter.kt
package com.futuretech.nfmovies.Adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.futuretech.nfmovies.Entity.Movie
import com.futuretech.nfmovies.Listener.MovieItemClickListener
import com.futuretech.nfmovies.R
import com.futuretech.nfmovies.Utils.ImageUtil

class MovieRecyclerViewAdapter(private val context: Context?, private val movies: List<Movie>, internal var movieItemClickListener: MovieItemClickListener) : RecyclerView.Adapter<MovieRecyclerViewAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.movie_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.name.text = movies[position].name
        ImageUtil.display(context!!, movies[position].img, holder.img, null)
    }

    override fun getItemCount(): Int {
        return movies.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val name: TextView
        val img: ImageView

        init {
            name = itemView.findViewById(R.id.movie_name)
            img = itemView.findViewById(R.id.movie_img)
            itemView.setOnClickListener { movieItemClickListener.onMovieClick(movies[adapterPosition], img) }
        }
    }
}
